### DISCUSSION TITLE
Params key not in get_artist

### DISCUSSION THREAD

**Discussion by kalebbroo** [sigma67/ytmusicapi#388] [Q&A] [answered]
**Describe the bug**
I am trying to return a list of all the albums from an artist using get_artist_albums. I cannot figure out how to get the perams key. I am only just learning how to code so this is most likely just a dumb mistake on my part. The api docs says to get params use get_artist but I am not able to return that key. 

Error searching for artist albums: 'params'
Traceback (most recent call last):
  File "C:\Users\kaleb\Desktop\projects\search_youtube\embed.py", line 46, in search_albums
    artist_albums_params = artist_data['albums']['params']
                           ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
KeyError: 'params'

**To Reproduce**
I plan to make this a discord bot, I search for an artist I have been using "will smith" and "loona" for testing. The code will return an key error.

Here is my code: 
async def search_albums(ctx, artist_name: str):
    artist_name = artist_name.title()
    print(f"Searching for albums by artist: {artist_name}")

    ytmusic = YTMusic()

    try:
        search_result = ytmusic.search(artist_name, filter='artists', limit=1)
        print(f"Search Result: {search_result}")

        if not search_result:
            await ctx.send(f"No artist found for '{artist_name}'")
            return

        artist_id = search_result[0]['browseId']
        artist_data = ytmusic.get_artist(artist_id)
        print(f"Artist Data First: {artist_data}")
        artist_name = artist_data.get('name')
        artist_channel_id = artist_data.get('channelId')
        artist_data = ytmusic.get_artist(artist_channel_id)
        print(f"Artist Data: {artist_data}")
        print(f"Channel ID: {artist_channel_id}")
        artist_albums_params = artist_data['albums']['params']
        album_data = ytmusic.get_artist_albums(artist_channel_id, artist_albums_params)

**Additional context**
If this is something I am doing wrong in my code just let me know. I have been trying to figure this out all day :(

**Comment by sigma67** [comment, marked as answer]
Hi, this doesn't work for all artists. Notably when there is no "More" button on the artist's page next to the albums, there won't be any params.

For example, for [Will Smith](https://music.youtube.com/channel/UCdHscQsCBWSfcW50NjdJWkA), singles would work but albums would not:

![image](https://user-images.githubusercontent.com/16363825/235438053-20f664c6-cfe4-4174-8f68-2b87e5c71276.png)

**Comment by kalebbroo** [comment]
Ah I see now, thanks for the response. I just added this to make my code work when there is no perams key. Seems to be holding up.

`if artist_albums_params:
            album_data = ytmusic.get_artist_albums(artist_channel_id, artist_albums_params)
        else:
            album_data = artist_data['albums']['results']`

**Reply by BrianZbr** [reply]
`artist_data['albums']['results']` seems to be limited to 10 results though... or is that just me?

**Reply by sigma67** [reply]
See the artist page on YTM Web, it will never show more than 10 for `get_artist`

Full list is only available via `get_artist_albums`

**Reply by BrianZbr** [reply]
I see, thanks... So I'm trying that with code based on what the OP provided, but this gives me `KeyError: 'musicCarouselShelfRenderer'`. The artist in this case _does_ have the More button.

```
from ytmusicapi import YTMusic
ytmusic = YTMusic()
artist_name = "autechre"

search_result = ytmusic.search(artist_name, filter='artists')
artist_data = ytmusic.get_artist(search_result[0]['browseId'])
artist_channel_id = artist_data.get('channelId')
artist_albums_params = artist_data['albums']['params']

album_data = ytmusic.get_artist_albums(artist_channel_id, artist_albums_params)
```

I also noticed that the `limit` function used by the OP doesn't actually seem to do anything?

**Reply by kalebbroo** [reply]
At the time I made that I was about 2 months into learning python. I would not trust anything I wrote at that time 😂. 

I do remember it working for my discord bot and I was not displaying more than 20 results anyway. 

If you want to look at my embarrassing first project to see what I did look at my repo plex_music_bot.
